src/Contrib/Jaeger/IdConverter.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Jaeger;
+
+use Brick\Math\BigInteger;
+
+/**
+ * An implementation of the section of the spec around converting OTEL's span/trace id representations to Thrift's format
+ * https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/jaeger.md#ids
+ */
+class IdConverter
+{
+    public static function convertOtelToJaegerTraceIds(string $traceId): array
+    {
+        $traceIdLow = self::convertHexStringToSignedInt(substr($traceId, 0, 16));
+        $traceIdHigh = self::convertHexStringToSignedInt(substr($traceId, 16, 32));
+
+        return [
+            'traceIdLow' => $traceIdLow,
+            'traceIdHigh' => $traceIdHigh,
+        ];
+    }
+
+    public static function convertOtelToJaegerSpanId(string $spanId): int
+    {
+        return self::convertHexStringToSignedInt($spanId);
+    }
+
+    private static function convertHexStringToSignedInt(string $hexString): int
+    {
+        //This is using high precision arithmetic because PHP by default won't do subtraction correctly for large integers (e.g. (2^63 - 1) - (2^63))
+        $hexStringAsInteger = BigInteger::fromBase($hexString, 16);
+
+        $shiftedInteger = $hexStringAsInteger->minus(BigInteger::fromBase('8000000000000000', 16)); //The number being subtracted is 2^63
```

Comment:
could it be written as `pow(2, 63)` for ease of reading?

Reply:
I updated the comments around this to try and clarify, but I don't think integers in PHP can support 2^63, so pow(2,63) probably won't work directly.

There may be other ways to represent it using the brick\math library though but nothing is catching my eye from another look at their READme - https://github.com/brick/math

Reply:
Maybe defining a constant would be appropriate here?

Reply:
Yeah that seems fine to me, updated.

Not in love with my initial naming so all suggestions are welcome

Reply:
(I'm learning about big numbers as I go, here)... So, it's really `PHP_INT_MAX+1`, right?
```php
php > var_dump(dechex(PHP_INT_MAX+1));
string(16) "8000000000000000"
```
Is it reliable/efficient to just calculate it, as above? As to constant name, `2_POW_63_HEX` ? Too brief?

Reply:
Yup about PHP_INT_MAX+1 (for 64-bit which we're requiring elsewhere iirc)

And whaaaa about that example and the [docs for dechex](https://www.php.net/manual/en/function.dechex.php) that say "The largest number that can be converted is PHP_INT_MAX * 2 + 1"!?! What does that even mean, like is PHP not representing PHP_INT_MAX * 2 + 1 as an integer or something? I wonder if it's getting converted to a float and then dechex is somehow still handling that passably 

(EDIT: yeah when I print PHP_INT_MAX in an online REPL I get 9223372036854775807 but when I print PHP_INT_MAX+1 I get 9.2233720368548E+18. And yeah when I print gettype(PHP_INT_MAX+1) it outputs "double", so that might be what's going on

So I guess it could be computed with pow(2,63), but that'll end up as a double and then we probably can't subtract properly between brick\math's BigInteger and BigDecimal representations (the latter which this would probably have to use if I had to guess) is what I would expect at least.

Also no preference on the constant name. Just lmk and I can switch whatever in.